aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -50,6 +54,7 @@
   private transient S3Client client;
   private MetricsContext metrics = MetricsContext.nullMetrics();
   private final AtomicBoolean isResourceClosed = new AtomicBoolean(false);
+  private List<Tag> tags;
```

Comment:
should this be a set instead? Could you check if the object hashcode is implemented?

Reply:
also similar comment, should be named `writeTags`

Reply:
Used a `Set`. Yes, object hashcode is implemented. Also, I have renamed to `writeTags`.

Reply:
Used a `Set`. Yes, object hashcode is implemented. Also, I have renamed to `writeTags`.